lib/SILOptimizer/LoopTransforms/LICM.cpp

```diff
@@ -887,6 +902,10 @@ void LoopTreeOptimization::analyzeCurrentLoop(
   for (auto *BB : Loop->getBlocks()) {
     SmallVector<SILInstruction *, 8> sideEffectsInBlock;
     for (auto &Inst : *BB) {
+      if (hasOwnershipOperandsOrResults(&Inst)) {
```

Comment:
Can you ensure there are tests covering other ownership instructions without side-effects ? load_borrow/destructure_struct/destructure_tuple etc

Reply:
I added a few tests - for instructions which really don't work if hoisted out of the loop.
This check is currently overly conservative. E.g. it is possible to hoist guaranteed values out of the loop.
Let's improve that when we convert the whole LICM optimization to OSSA.